src/Orleans/Providers/ILeaseProvider.cs

```diff
@@ -111,4 +150,9 @@ public interface ILeaseProvider
         Task Release(string category, AcquiredLease[] aquiredLeases);
     }
 
+    public class ResourceCategory
```

Comment:
I don't think we want the categories as part of the interface definition.  I suggest each category be defined by the system responsible for that category.  So the queue balancer would define StreamQueues category, for instance.